### DISCUSSION TITLE
What's the thought process behind limiting getServerSideProps to pages only?

### DISCUSSION THREAD

**Discussion by abhinav-anshul** [vercel/next.js#36919] [Help]
Hi,

Why `getServerSideProps` is limited to the pages folder and can't be used in other folders such as components? If the components getting rendered on the page anyhow?

**Comment by GabenGar** [comment]
Because pages folder is used to build the file system based routing.

**Reply by icyJoseph** [reply]
This, in addition to the fact that React rendering was/is (hard to tell these days), synchronous.

How should the rendering model look like when you ought to render data, which is partially/fully asynchronous? React has proposed a solution for that called Suspense Mode, which will compound with Server Components.

However, going straight to your question, the reason is, again, that `getServerSideProps` is asynchronous, and injecting it at component level means handling, awaiting for the data, handling errors, caching, revalidating, etc.

These issues are present at page level, but at page level there's only one child (the entire page), and the `.json` file is first retrieved, and then the page is rendered. Going directly to a page is a noop, in this sense.

Notice that nothing prohibits you from calling an `api/route`, to get data on the server, and then use that as props to a component. The issue at hand is, how to define `gSSP` at component level, so that, similarly to how styles, markup and logic, live together, so does the data fetching, but from the server.

(Also they'd have to scan every single component looking for an export named `gSSP`, and handled all of the stuff that implies)
